packages/styled-components/transforms/src/visitors/transpile_css_prop/transpile.rs

```diff
@@ -26,6 +30,25 @@ struct TranspileCssProp {
     injected_nodes: Vec<Stmt>,
 
     identifier_idx: usize,
+    styled_idx: HashMap<JsWord, usize>,
+    bindings: Vec<AHashSet<JsWord>>,
+}
+
+impl TranspileCssProp {
+    fn next_styled_idx(&mut self, key: JsWord) -> usize {
+        let idx = self.styled_idx.entry(key).or_insert(0);
+        *idx += 1;
+        *idx
+    }
+    fn is_top_level_ident(&mut self, ident: &Ident) -> bool {
```

Comment:
Comparing the `ident.span.ctxt` with `top_level_ctxt` is enough for this operation.